halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/ClouddriverCachingProfileFactory.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentConfiguration;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerArtifact;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerRuntimeSettings;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ClouddriverCachingProfileFactory extends StringBackedProfileFactory {
+  @Override
+  protected String getRawBaseProfile() {
+    return "";
+  }
+
+  @Override
+  protected void setProfile(Profile profile, DeploymentConfiguration deploymentConfiguration,
```

Comment:
I still think this should be checked into the clouddriver repo under `./halconfig/clouddriver-caching.yml`

Reply:
The advantage being, we don't couple the halyard version (which contains this string) to the clouddriver version (which may require different/new properties in the future). By checking it into the clouddriver repo you ensure that the properties always match clouddriver's config requirements.

Reply:
That makes sense. Can I leave that as a TODO? I can handle adding ```clouddriver-caching.yml```, etc to their respective repos once they are finalized.